### PR TITLE
fix (fcm): replaced global.wretch by fetch in registerFCMTopic

### PR DESCRIPTION
**Pull request by supreeth-spiger**
fix made to receive token from api call using **fetch**. Since **global.wretch** is not present in all existing publishers

### CODE REVIEW

File: client/impl/fcm.js [hunk 1]
```diff
@@ -32,11 +32,18 @@ function updateToken(firebaseInstance) {
     .then(token => {
       return registerFCMTopic(token);
     })
-    .catch(() => {
+    .catch(err => {
+      console.log(err, "Could not retrieve token from firebase");
       throw new Error("Could not retrieve token from firebase");
```
Comment (asimqt) [inline client/impl/fcm.js]: why this log?

File: client/impl/fcm.js [hunk 2]
```diff
@@ -32,11 +32,18 @@ function updateToken(firebaseInstance) {
...
 function registerFCMTopic(token) {
-  return global.wretch("/register-fcm-topic").post({ token: token });
+  return fetch("/register-fcm-topic", {
+    method: "post",
+    headers: {
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify({ token: token })
+  });
 }
```
Comment (asimqt) [inline client/impl/fcm.js]: 👍 is there any other place where we're using wretch?

File: client/impl/fcm.js (hunk 1 above)
Comment (supreeth-spiger) [reply client/impl/fcm.js]: to find the **Errors inside Promises** when promises are rejected on the client side.

Comment (asimqt) [reply client/impl/fcm.js]: We’re throwing an error just after on line 37. Isn’t that enough?

File: client/impl/fcm.js (hunk 2 above)
Comment (supreeth-spiger) [reply client/impl/fcm.js]: have cross checked , No where is the repo wretch is used.

File: client/impl/fcm.js (hunk 1 above)
Comment (supreeth-spiger) [reply client/impl/fcm.js]: got it , have made the changes

**Pull request merged**